portfolio/src/main/webapp/scripts/snake/sketch.js

```diff
@@ -0,0 +1,283 @@
+        y: 0, 
+        xConstraint1: 0, 
+        xConstraint2: halfWidth - GRID_SCALE, 
+        yConstraint1: 0, 
+        yConstraint2: halfHeight - GRID_SCALE, 
+        gridScale: GRID_SCALE, 
+        red: 255, 
+        green: 0, 
+        blue: 0
+    });
+    sec2.snake = new Snake({
+        x: halfWidth, 
+        y: 0, 
+        xConstraint1: halfWidth, 
+        xConstraint2: cnvDiv.offsetWidth - GRID_SCALE, 
+        yConstraint1: 0, 
+        yConstraint2: halfHeight - GRID_SCALE, 
+        gridScale: GRID_SCALE, 
+        red: 0, 
+        green: 255, 
+        blue: 0
+    });
+    sec3.snake = new Snake({
+        x: 0, 
+        y: halfHeight, 
+        xConstraint1: 0, 
+        xConstraint2: halfWidth - GRID_SCALE,
+        yConstraint1: halfHeight,
+        yConstraint2: cnvDiv.offsetHeight - GRID_SCALE,
+        gridScale: GRID_SCALE, 
+        red: 255, 
+        green: 0, 
+        blue: 255
+    });
+    sec4.snake = new Snake({
+        x: halfWidth, 
+        y: halfHeight, 
+        xConstraint1: halfWidth, 
+        xConstraint2: cnvDiv.offsetWidth - GRID_SCALE,
+        yConstraint1: halfHeight,
+        yConstraint2: cnvDiv.offsetHeight - GRID_SCALE,
+        gridScale: GRID_SCALE, 
+        red: 255, 
+        green: 100, 
+        blue: 100
+    });
+
+    // Initialize new food in each section based on the section's constraints 
+    sec1.food = new Food({
+        xConstraint1: 0, 
+        xConstraint2: halfWidth - GRID_SCALE,
+        yConstraint1: 0,
+        yConstraint2: halfHeight - GRID_SCALE,
+        gridScale: GRID_SCALE,
+        section: 1
+    });
+    sec2.food = new Food({
+        xConstraint1: halfWidth, 
+        xConstraint2: cnvDiv.offsetWidth - GRID_SCALE,
+        yConstraint1: 0,
+        yConstraint2: halfHeight - GRID_SCALE,
+        gridScale: GRID_SCALE,
+        section: 2
+    });
+    sec3.food = new Food({
+        xConstraint1: 0, 
+        xConstraint2: halfWidth - GRID_SCALE,
+        yConstraint1: halfHeight,
+        yConstraint2: cnvDiv.offsetHeight - GRID_SCALE,
+        gridScale: GRID_SCALE,
+        section: 3
+    });
+    sec4.food = new Food({
+        xConstraint1: halfWidth, 
+        xConstraint2: cnvDiv.offsetWidth - GRID_SCALE,
+        yConstraint1: halfHeight,
+        yConstraint2: cnvDiv.offsetHeight - GRID_SCALE,
+        gridScale: GRID_SCALE,
+        section: 4
+    });
+}
+
+/*
+ * Resize the canvas and initialize the game again when the window gets resized.
+ * Makes the game grid layout dynamic and ensures that the game can be played on 
+ * different screen sizes comfortably.
+ */
+function windowResized() {
+    cnv = resizeCanvas(floor(cnvDiv.offsetWidth), floor(cnvDiv.offsetHeight));
+    initializeGame();
+}
+
+/*
+ * Draw is continuously called to update the game. Detects changes in game
+ * state, like game difficulty, snake death, snake eating, or key press to
+ * change direction of snake. Updates every snake on call.
+ */
+function draw() {
```

Comment:
You might consider extracting some of what's going on here into helper functions.  The comments are great but still could be a bit more readable if this method were a little less bloated and you could rely on calls to helper methods to understand the game flow in a more condensed manner.

Reply:
This is an excellent point, thanks for bringing it up because I know we've talked about extracted things out to helper functions before in a 1:1 call. It makes sense that instead of making bloated functions with lots of comments to instead extract things that need lots of commenting into helper functions with very verbose function names, so there's no need for a separate comment when the helper function is called.

I pushed a change for this, let me know if it looks ok or if there's anything you'd change!

Reply:
Yes great job extracting out into helper functions!  

Supernit: you can totally leave this but something to consider for the future-- in case I accidentally mislead you, I think it's useful to have *descriptive* function names for sure so that you don't need to do inline commenting whenever a helper is called within a larger function, but I think slightly less verbose function names are actually preferred :)

Reply:
Awesome thanks for the feedback Colette! I think as I keep practicing what you're saying, my function and variable naming abilities will improve. It seems like the perfect names are those that are just descriptive enough that you know what they're doing at a high level without a comment, but at the same time succinct in a manner that extra words don't bloat the meaning. I'll keep working at it!